Description: added fullscreen
 added fullscreen function, to go fullscreen and back, with F key
Author: Nicolas HOUDELOT <nicolas@demosdebs.org>
Applied-Upstream: no
Last-Update: 2024-03-28
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/intro.cpp
+++ b/src/intro.cpp
@@ -3150,6 +3150,13 @@
   }
 }
 
+void ToggleFullscreen(SDL_Window* Window) {
+    int FullscreenFlag = SDL_WINDOW_FULLSCREEN;
+    int isFullscreen = SDL_GetWindowFlags(Window) & FullscreenFlag;
+    SDL_SetWindowFullscreen(Window, isFullscreen ? 0 : FullscreenFlag);
+    SDL_ShowCursor(isFullscreen);
+}
+
 void GlobalContainer::fillState(State *op) const
 {
   Vector3 camera_pos = op->getPosition();
@@ -4395,6 +4402,10 @@
               time_delta = 0;
               break;
 
+			case SDLK_f:
+              ToggleFullscreen(sdlWindow);
+              break;
+
             default:
               break;
           }
